scratch_space/README.md

```diff
@@ -0,0 +1,55 @@
+# [![Build Status](https://travis-ci.org/dart-lang/build.svg?branch=master)](https://travis-ci.org/dart-lang/build)
+
+## [`ScratchSpace`][dartdoc:ScratchSpace]
+
+A `ScratchSpace` is a thin wrapper around a temporary directory. The
+constructor takes zero arguments, so making one is as simple as doing
+`new ScratchSpace()`.
+
+### Adding assets to a `ScratchSpace`
+
+To add assets to the `ScratchSpace`, you use the `ensureAssets` method, which
+takes an `Iterable<AssetId>` and an `AssetReader` (for which you should
+generally pass your `BuildStep` which implements that interface).
+
+**Note:** It is important to note that the `ScratchSpace` does not guarantee
+that the version of a file within it is the most updated version, only that
+some version of it exists. For this reason you should create a new
+`ScratchSpace` for each build (within a single build files cannot change, but
+between builds they might).
+
+### Deleting a `ScratchSpace`
+
+When you are done with a `ScratchSpace` you should call `delete` on it to make
```

Comment:
Can we look for a way for this to be done automatically using new build lifecycle apis in package:build? Will we try to do that in a followup?

Reply:
Yes, once those are available we will add some helpers here that use them.